services/autoscaling/tests/unit/conftest.py

```diff
@@ -40,12 +75,385 @@ def app_environment(
+                ), f"service {found_service['Spec']['Name']}'s task is {service_task['Status']['State']}"
+                print(
+                    f"<-- service {found_service['Spec']['Name']} is now {service_task['Status']['State']} {'.'*number_of_success}"
+                )
+                number_of_success += 1
+                assert (number_of_success * WAIT_TIME) >= SUCCESS_STABLE_TIME_S
+                print(
+                    f"<-- service {found_service['Spec']['Name']} is now {service_task['Status']['State']} after {SUCCESS_STABLE_TIME_S} seconds"
+                )
+
+    return _runner
+
+
+@pytest.fixture(scope="module")
+def mocked_aws_server() -> Iterator[ThreadedMotoServer]:
+    """creates a moto-server that emulates AWS services in place
+    NOTE: Never use a bucket with underscores it fails!!
+    """
+    server = ThreadedMotoServer(ip_address=get_localhost_ip(), port=unused_port())
+    # pylint: disable=protected-access
+    print(f"--> started mock AWS server on {server._ip_address}:{server._port}")
+    print(
+        f"--> Dashboard available on [http://{server._ip_address}:{server._port}/moto-api/]"
+    )
+    server.start()
+    yield server
+    server.stop()
+    print(f"<-- stopped mock AWS server on {server._ip_address}:{server._port}")
+
+
+@pytest.fixture
+def reset_aws_server_state(mocked_aws_server: ThreadedMotoServer) -> Iterator[None]:
+    # NOTE: reset_aws_server_state [http://docs.getmoto.org/en/latest/docs/server_mode.html#reset-api]
+    yield
+    # pylint: disable=protected-access
+    requests.post(
+        f"http://{mocked_aws_server._ip_address}:{mocked_aws_server._port}/moto-api/reset",
+        timeout=10,
+    )
+
+
+@pytest.fixture
+def mocked_aws_server_envs(
+    app_environment: EnvVarsDict,
+    mocked_aws_server: ThreadedMotoServer,
+    reset_aws_server_state: None,
+    monkeypatch: pytest.MonkeyPatch,
+) -> Iterator[EnvVarsDict]:
+    changed_envs = {
+        "EC2_ENDPOINT": f"http://{mocked_aws_server._ip_address}:{mocked_aws_server._port}",  # pylint: disable=protected-access
+        "EC2_ACCESS_KEY_ID": "xxx",
+        "EC2_SECRET_ACCESS_KEY": "xxx",
+    }
+    yield app_environment | setenvs_from_dict(monkeypatch, changed_envs)
+
+
+@pytest.fixture
+def aws_allowed_ec2_instance_type_names(
+    app_environment: EnvVarsDict,
+    monkeypatch: pytest.MonkeyPatch,
+) -> Iterator[EnvVarsDict]:
+    changed_envs = {
+        "EC2_INSTANCES_ALLOWED_TYPES": json.dumps(
+            [
+                "t2.xlarge",
+                "t2.2xlarge",
+                "r5n.4xlarge",
+                "r5n.8xlarge",
+            ]
+        ),
+    }
+    yield app_environment | setenvs_from_dict(monkeypatch, changed_envs)
+
+
+@pytest.fixture
+def aws_vpc_id(
+    mocked_aws_server_envs: None,
+    app_environment: EnvVarsDict,
+    monkeypatch: pytest.MonkeyPatch,
+    ec2_client: EC2Client,
+) -> Iterator[str]:
+    vpc = ec2_client.create_vpc(
+        CidrBlock="10.0.0.0/16",
```

Comment:
Could you have this as a config-var at the top of the `py` file? This would make this hardcoded thingy less deeply burried in the test code. I guess this comes from @Surfict original version, but this deeply hidden hardcoded magic Cidr blok might bite us in the ass in the distance future if the Cidr block on our actual aws vpc changes for any reason ;)

Reply:
...or simply as a pytest fixture so we can add more CIDR variations later

Reply:
I agree I can do a fixture here. will do that.
BUT, there were no tests originaly they were all created with this PR.

Also bear in mind, that the tests are not running against the real AWS but a mock library called Moto.
Therefore at the moment I am not sure these CIDR play any role here but to allow me to create a VPC in the Moto instance.